### DISCUSSION TITLE
How to control the video length ?

### DISCUSSION THREAD

**Discussion by SIFOR-GAMES** [nateraw/stable-diffusion-videos#136] [General]
Hello everyone sorry for the noob question may I know how is it possible to control the video length of the prompt result?

**Comment by honeybansal9** [comment]
@SIFOR-GAMES to change the video length you need to increase the num_interpolation_steps so if your num_interpolation_steps=5 then your video length will be 1 sec and you can increase it in the multiples of 5. This is how I discovered by changing num_interpolation_steps=30 the video length gets changed to 6secs.

**Reply by SIFOR-GAMES** [reply]
Thanks

**Comment by nateraw** [comment]
@honeybansal9 is on the right track but here's a more in depth explaination...

`num_interpolation_steps` - the number of frames to generate in between each prompt. **The default is 5.** If its an int, it gets turned into a list that's one item shorter than the provided `prompts`/`seeds`.  Heres the line that does that:

https://github.com/nateraw/stable-diffusion-videos/blob/6162a04def04f298f52f80c2a7d87da8a8b1ddb4/stable_diffusion_videos/stable_diffusion_pipeline.py#L768

the `fps` parameter (which defaults to 30), is the frames per second of the output video we generate.

If you were to divide the sum of `num_interpolation_steps` list by fps, you'd get the duration.

Now that we know that, and we have some desired duration (in this case 5 seconds) and fps (using the default of 30), you could do something like this...

```python
prompts = ['a cat', 'a dog', 'a horse']
seeds = [42, 1234, 5321]
desired_duration = 5.0
fps = 30

total_frames_to_generate = desired_duration * fps
# >>> 150.0
num_interpolation_steps = int(total_frames_to_generate // (len(prompts) - 1))
# >>> 75
# this means 75 frames between 'a cat' and 'a dog'
# then 75 frames between 'a dog' and 'a horse'

# Then use the computed value to generate your video
pipe.walk(prompts, seeds, num_interpolation_steps=num_interpolation_steps, fps=fps)
```

hope this helps 😄

**Reply by SIFOR-GAMES** [reply]
Thanks got it... appreciate it !

**Comment by nitrocon** [comment]
And this is how it looks in a working script:

from stable_diffusion_videos import StableDiffusionWalkPipeline
import torch

pipeline = StableDiffusionWalkPipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4",
    torch_dtype=torch.float16,
).to("cuda")

fps = 30  # Frames per second
video_length_in_seconds = 10  

num_interpolation_steps = fps * video_length_in_seconds

video_path = pipeline.walk(
    prompts=['a cat', 'a dog'],
    seeds=[42, 1337],
    num_interpolation_steps=num_interpolation_steps, 
    height=1024,  
    width=1024,   
    output_dir='dream',  
    name='animals_test', 
    guidance_scale=6,  
    num_inference_steps=100,  
)

print(f"Video saved as .mp4 in: {video_path}")
